lib/active_record/tasks/sqlserver_database_tasks.rb

```diff
@@ -69,10 +67,24 @@ def structure_dump(filename)
         dump.gsub!(/nvarchar\(-1\)/, 'nvarchar(max)')         # Fix nvarchar(-1) column defs
         dump.gsub!(/text\(\d+\)/, 'text')                     # Fix text(16) column defs
         File.open(filename, "w") { |file| file.puts dump }
+
+        # defncopy appears to truncate definition output in some circumstances
+        # Also create view needs to be the first operation in the batch.
+        File.open(filename, 'a') { |file|
+          connection.select_all("select definition, o.type from sys.objects as o join sys.sql_modules as m on m.object_id = o.object_id where o.type = 'V'").each do |row|
+            file.puts "\r\nGO\r\n#{row['definition']}"
```

Comment:
Why use this SQL vs our existing `view_information` method? Does one work better than the other?